cmd/bclient/bserver/chunkfile.go

```diff
@@ -0,0 +1,58 @@
+package bserver
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+const BogusFileId string = "Uninitialized"
+
+func chunkAndUpload(fileroot string, srcFile string, srcFileMd5 []byte, item string, fileChunkSize int) (string, error) {
+
+	sourceFile, err := os.Open(path.Join(fileroot, srcFile))
+
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer sourceFile.Close()
+
+	chunk := make([]byte, fileChunkSize)
+
+	var fileId string = BogusFileId
+
+	// upload the chunk
+
+	for chunkNo := int64(1); chunkNo > 0; chunkNo++ {
+		bytesRead, readErr := sourceFile.Read(chunk)
+
+		if bytesRead > 0 {
+
+			//filename := chunkFileName
+
+			chMd5 := md5.Sum(chunk[:bytesRead])
+
+			fileId, err = PostUpload(chunk[:bytesRead], chMd5[:], srcFileMd5, fileId)
+
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+
+			continue
+		}
+
+		if readErr != nil && readErr != io.EOF {
```

Comment:
This one you'll have to live with.

On Fri, Jan 22, 2016 at 2:12 PM, Don Brower <notifications@github.com>
wrote:

> In cmd/bclient/bserver/chunkfile.go
> <https://github.com/ndlib/bendo/pull/56#discussion_r50577320>:
>
> > +		if bytesRead > 0 {
> > +
> > +			//filename := chunkFileName
> > +
> > +			chMd5 := md5.Sum(chunk[:bytesRead])
> > +
> > +			fileId, err = PostUpload(chunk[:bytesRead], chMd5[:], srcFileMd5, fileId)
> > +
> > +			if err != nil {
> > +				fmt.Println(err.Error())
> > +			}
> > +
> > +			continue
> > +		}
> > +
> > +		if readErr != nil && readErr != io.EOF {
>
> try to do all the error handling before the work. In this case that means,
> checking for errors before checking whether bytesRead is > 0. Then the
> code doing the work (inside the bytesRead > 0 block) won't be nested and
> you won't need the continue statement.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ndlib/bendo/pull/56/files#r50577320>.
>